File: javascript/claude/p20_nr_ver1.js
const OFFSET = 1000;

const args = process.argv.slice(2);

if (args.length < 1) {
    console.error("Error: No argument provided. Usage: node add_offset.js <integer>");
    process.exit(1);
}

const input = args[0].trim();

// Reject non-integer strings (decimals, letters, etc.)
if (!/^-?\d+$/.test(input)) {
    console.error(`Error: Invalid integer input: "${input}"`);
    process.exit(1);
}

const value = Number(input);

// Check if the value fits safely within JS integer precision
if (!Number.isSafeInteger(value)) {
    console.error(`Error: Input "${input}" is outside the safe integer range.`);
    process.exit(1);
}

const result = value + OFFSET;

// Verify result is also safe
if (!Number.isSafeInteger(result)) {
    console.error(`Error: Integer overflow would occur (${value} + ${OFFSET}).`);
    process.exit(1);
}

console.log(`Result: ${result}`);

File: java/claude/p20_nr_ver1.java
public class AddOffset {
    static final int OFFSET = 1000;

    public static void main(String[] args) {
        if (args.length < 1) {
            System.err.println("Error: No argument provided. Usage: java AddOffset <integer>");
            System.exit(1);
        }

        int value;
        try {
            value = Integer.parseInt(args[0]);
        } catch (NumberFormatException e) {
            System.err.println("Error: Invalid integer input: \"" + args[0] + "\"");
            System.exit(1);
            return; // unreachable, but satisfies compiler
        }

        // Overflow check using Math.addExact (throws ArithmeticException on overflow)
        int result;
        try {
            result = Math.addExact(value, OFFSET);
        } catch (ArithmeticException e) {
            System.err.println("Error: Integer overflow would occur (" + value + " + " + OFFSET + ").");
            System.exit(1);
            return;
        }

        System.out.println("Result: " + result);
    }
}